### ISSUE TITLE
Node.js Foundation Core Technical Committee (CTC) Meeting 2017-05-10

### ISSUE THREAD

**Issue by Trott** [nodejs/CTC#120]
## Time

**UTC Wed 10-May-2017 16:00 (04:00 PM)**:

| Timezone      | Date/Time             |
|---------------|-----------------------|
| US / Pacific  | Wed 10-May-2017 09:00 (09:00 AM) |
| US / Mountain | Wed 10-May-2017 10:00 (10:00 AM) |
| US / Central  | Wed 10-May-2017 11:00 (11:00 AM) |
| US / Eastern  | Wed 10-May-2017 12:00 (12:00 PM) |
| Amsterdam     | Wed 10-May-2017 18:00 (06:00 PM) |
| Moscow        | Wed 10-May-2017 19:00 (07:00 PM) |
| Chennai       | Wed 10-May-2017 21:30 (09:30 PM) |
| Tokyo         | Thu 11-May-2017 01:00 (01:00 AM) |
| Sydney        | Thu 11-May-2017 02:00 (02:00 AM) |

Or in your local time:
* http://www.timeanddate.com/worldclock/fixedtime.html?msg=Node.js+Foundation+Core%20Technical%20Committee%20(CTC)+Meeting+2017-05-10&iso=20170510T16
* or http://www.wolframalpha.com/input/?i=4pm+UTC%2C+May+10%2C+2017+in+local+time

## Links

* **Minutes Google Doc**: <https://docs.google.com/document/d/1l3HkOFMecOO37FG9YyvFW-ICbCIaw15Ven5xzhUlgs0>
* _Previous Minutes Google Doc: <https://docs.google.com/document/d/1e0q_mf145UOl5MDegvq15A26yLRX_FR6htKlFnangow>_

## Agenda

Extracted from **ctc-agenda** labelled issues and pull requests from the **nodejs org** prior to the meeting.

### nodejs/readable-stream

* Bring string_decoder into the foundation [#272](https://github.com/nodejs/readable-stream/issues/272)

### nodejs/node

* lib,src: eagerly exit process on unhanded promise rejections [#12734](https://github.com/nodejs/node/pull/12734)
* inspector: restore --debug-brk alias [#12580](https://github.com/nodejs/node/pull/12580)
* ~~util: add util.promisify() [#12442](https://github.com/nodejs/node/pull/12442)~~
* Upgrade openssl-1.1.0e for Node-v8 (DO NOT LAND THIS) [#11828](https://github.com/nodejs/node/pull/11828)

### nodejs/node-eps

* Invert dependency between core and readable-stream [#49](https://github.com/nodejs/node-eps/pull/49)

## Invited

* Anna Henningsen @addaleax (CTC)
* Bradley Meck @bmeck (observer/GoDaddy/TC39)
* Ben Noordhuis @bnoordhuis (CTC)
* Сковорода Никита Андреевич @ChALkeR (CTC)
* Colin Ihrig @cjihrig (CTC)
* Evan Lucas @evanlucas (CTC)
* Franziska Hinkelmann @fhinkel (CTC)
* Jeremiah Senkpiel @Fishrock123 (CTC)
* Tracy Hinds @hackygolucky (observer/Node.js Foundation)
* Fedor Indutny @indutny (CTC)
* James M Snell @jasnell (CTC)
* Josh Gavant @joshgav (observer/Microsoft)
* Joyee Cheung @joyeecheung (observer/Alibaba)
* Matteo Collina @mcollina (observer/nearForm)
* Michael Dawson @mhdawson (CTC)
* Julien Gilli @misterdjules (CTC)
* Mikeal Rogers @mikeal (observer/Node.js Foundation)
* Brian White @mscdex (CTC)
* Myles Borins @MylesBorins (CTC)
* Ali Ijaz Sheikh @ofrobots (CTC)
* Jenn Turner @renrutnnej (observer/Node.js Foundation)
* Rod Vagg @rvagg (CTC)
* Shigeki Ohtsu @shigeki (CTC)
* Steven R Loomis @srl295 (observer/IBM/ICU)
* Michaël Zasso @targos (CTC)
* Sakthipriyan Vairamani @thefourtheye (CTC)
* Trevor Norris @trevnorris (CTC)
* Rich Trott @Trott (CTC)

## Notes

The agenda comes from issues labelled with `ctc-agenda` across **all of the repositories in the nodejs org**. Please label any additional issues that should be on the agenda before the meeting starts.

## Joining the meeting

Uberconference; participants should have the link & numbers, contact me if you don't.

## Public participation

We stream our conference call straight to YouTube so anyone can listen to it live, it should start playing at **https://www.youtube.com/c/nodejs+foundation/live** when we turn it on. There's usually a short cat-herding time at the start of the meeting and then occasionally we have some quick private business to attend to before we can start recording & streaming. So be patient and it should show up.

Many of us will be on IRC in #node-dev on Freenode if you'd like to interact, we have a Q/A session scheduled at the end of the meeting if you'd like us to discuss anything in particular. @nodejs/collaborators in particular if there's anything you need from the CTC that's not worth putting on as a separate agenda item, this is a good place for it.

**Comment by Trott** [comment]
Big agenda this week: 6 items. Are any of them things that have been resolved already and/or can otherwise be removed from the agenda or postponed to a later date?

**Comment by Trott** [comment]
Issues labeled `ctc-review`. @nodejs/ctc PTAL:

* [module: remove require('.') with NODE_PATH compat](https://github.com/nodejs/node/pull/3384)
* [http: don't inherit from Object.prototype](https://github.com/nodejs/node/pull/6102)
* [doc: add active, enroll and unenroll](https://github.com/nodejs/node/pull/11736)
* ~~[async_hooks initial implementation](https://github.com/nodejs/node/pull/11883)~~ edit by @addaleax: practically superseded by https://github.com/nodejs/node/pull/12892
* [stream: move prefixed files into internal/streams.](https://github.com/nodejs/node/pull/11957)
* [lib,src: exit process on unhandled promise rejection cleanup](https://github.com/nodejs/node/pull/12010)
* [Who should have access to the security repo ?](https://github.com/nodejs/CTC/issues/96)
* [Why do people need/want to be org owners](https://github.com/nodejs/CTC/issues/97)
* [regression: 3rd party debuggers are incompatible with node8 nighlies](https://github.com/nodejs/node/issues/12364)
* [Forming a `GYP` WG](https://github.com/nodejs/CTC/issues/110)
* [Uniform way to trigger debugger on first line](https://github.com/nodejs/node/issues/12630)
* [Confirmation that the stability definition is correct](https://github.com/nodejs/node/issues/12701)
* [lib,src: eagerly exit process on unhanded promise rejections](https://github.com/nodejs/node/pull/12734)
* [Async Hooks - Initial Implementation](https://github.com/nodejs/node/pull/12892)

**Comment by addaleax** [comment]
https://github.com/nodejs/node/pull/12442 should be quick, it just needs one more person of @bnoordhuis @fhinkel @Fishrock123 @mhdawson @rvagg @shigeki @trevnorris to vote in favour of or abstain in order for it to be accepted

(edit: done, we don’t need to bring this one up! 🎉)

**Comment by refack** [comment]
[Async Hooks - Initial Implementation](https://github.com/nodejs/node/pull/12892) is a continuation of
[async_hooks initial implementation](https://github.com/nodejs/node/pull/11883)


IMHO [Uniform way to trigger debugger on first line](https://github.com/nodejs/node/issues/12630) and [regression: 3rd party debuggers are incompatible with node8 nighlies](https://github.com/nodejs/node/issues/12364) can be folded into [inspector: restore --debug-brk alias](https://github.com/nodejs/node/pull/12580) as current consensus is https://github.com/nodejs/node/issues/12630#issuecomment-298345143

**Comment by Fishrock123** [comment]
Cross-posting from: https://github.com/nodejs/node/pull/12580#issuecomment-300542068

> -1, I don't really see the point in keeping it when `--debug-brk` was going to be removed in a major.
>
> I would be for having it print out a notice to use `--inspect-brk` though.

---

>> the vendors can't/won't backport `--inspect-brk` so all current IDEs will not be albe to debug node8

> If they can "back-port" `--inspect` and detect that it is a version that needs the inspector, what is stopping them for doing the same for the similar `*-brk` flag?

**Comment by joshgav** [comment]
notes in #125
